util/cmpver/version_test.go

```diff
@@ -9,156 +9,189 @@ import (
-			want: -1,
-		},
+	{
+		name: "tailscale major",
+		v1:   "1.0-0",
+		v2:   "0.97-105",
+		want: 1,
+	},
+	{
+		name: "tailscale minor",
+		v1:   "0.98-0",
+		v2:   "0.97-105",
+		want: 1,
+	},
+	{
+		name: "tailscale patch",
+		v1:   "0.97-120",
+		v2:   "0.97-105",
+		want: 1,
+	},
+	{
+		name: "tailscale equal",
+		v1:   "0.97-105",
+		v2:   "0.97-105",
+		want: 0,
+	},
+	{
+		// TODO: this test passes because '-' < '.', not because we have
+		// more '.'-separated fields before the '-'. Is that what we want?
+		name: "tailscale weird extra field",
+		v1:   "0.96.1-0", // more fields == larger
+		v2:   "0.96-105",
+		want: 1,
+	},
+	{
+		// Though ۱ and ۲ both satisfy unicode.IsNumber, our previous use
+		// of strconv.ParseUint with these characters would have lead us to
+		// panic. We're now only looking at ascii numbers, so test these are
+		// compared as text.
+		name: "only ascii numbers",
+		v1:   "۱۱", // 2x EXTENDED ARABIC-INDIC DIGIT ONE
+		v2:   "۲",  // 1x EXTENDED ARABIC-INDIC DIGIT TWO
+		want: -1,
+	},
 
-		// A few specific OS version tests below.
-		{
-			name: "windows version",
-			v1:   "10.0.19045.3324",
-			v2:   "10.0.18362",
-			want: 1,
-		},
-		{
-			name: "windows 11 is everything above 10.0.22000",
-			v1:   "10.0.22631.2262",
-			v2:   "10.0.22000",
-			want: 1,
-		},
-		{
-			name: "android short version",
-			v1:   "10",
-			v2:   "7",
-			want: 1,
-		},
-		{
-			name: "android longer version",
-			v1:   "7.1.2",
-			v2:   "7",
-			want: 1,
-		},
-		{
-			name: "iOS version",
-			v1:   "15.6.1",
-			v2:   "15.6",
-			want: 1,
-		},
-		{
-			name: "Linux short kernel version",
-			v1:   "4.4.302+",
-			v2:   "4.0",
-			want: 1,
-		},
-		{
-			name: "Linux long kernel version",
-			v1:   "4.14.255-311-248.529.amzn2.x86_64",
-			v2:   "4.0",
-			want: 1,
-		},
-		{
-			name: "FreeBSD version",
-			v1:   "14.0-CURRENT",
-			v2:   "14",
-			want: 1,
-		},
-		{
-			name: "Synology version",
-			v1:   "Synology 6.2.4; kernel=3.10.105",
-			v2:   "Synology 6",
-			want: 1,
-		},
-	}
+	{
+		name: "zeroes equal",
+		v1:   "0001",
+		v2:   "01",
+		want: 0,
+	},
+	{
+		name: "zeroes nested equal",
+		v1:   "9.0001",
+		v2:   "009.01",
+		want: 0,
+	},
+	{
+		name: "chars",
+		v1:   "aaaa0",
+		v2:   "aaa0",
+		want: 1,
+	},
+
+	// A few specific OS version tests below.
+	{
+		name: "windows version",
+		v1:   "10.0.19045.3324",
+		v2:   "10.0.18362",
+		want: 1,
+	},
+	{
+		name: "windows 11 is everything above 10.0.22000",
+		v1:   "10.0.22631.2262",
+		v2:   "10.0.22000",
+		want: 1,
+	},
+	{
+		name: "android short version",
+		v1:   "10",
+		v2:   "7",
+		want: 1,
+	},
+	{
+		name: "android longer version",
+		v1:   "7.1.2",
+		v2:   "7",
+		want: 1,
+	},
+	{
+		name: "iOS version",
+		v1:   "15.6.1",
+		v2:   "15.6",
+		want: 1,
+	},
+	{
+		name: "Linux short kernel version",
+		v1:   "4.4.302+",
+		v2:   "4.0",
+		want: 1,
+	},
+	{
+		name: "Linux long kernel version",
+		v1:   "4.14.255-311-248.529.amzn2.x86 64",
```

Comment:
why this test change from underscore to space?

Reply:
Find and replace error - the correct usage of VSCode's "replace in selection" continues to evade me.